base/src/main/java/io/fabric8/launcher/base/http/HttpClient.java

```diff
@@ -47,10 +50,17 @@ public HttpClient() {
      * @param executorService used in the async methods
      */
     @Inject
-    public HttpClient(@Nullable ExecutorService executorService) {
-        client = createClient(executorService);
+    public HttpClient(final ExecutorService executorService) {
+        this.client = createClient(requireNonNull(executorService, "executorService must not be null"));
     }
 
+    private HttpClient(final OkHttpClient client) {
+        this.client = requireNonNull(client, "client must be specified.");
+    }
+
+    public static HttpClient createForTest() {
```

Comment:
I was thinking of lazily initializing the client (by having a private getClient() in here and using it in the methods implementation instead of the attribute directly)

Reply:
It's safer this way IMO

Reply:
I think it's easier to use the default constructor than to introduce a factory method here, it saves you from deprecating the default constructor :)

Reply:
I think it's really not easier. It requires a lazy client which is just required because of the proxy. Here I set the client to null for the proxy init, making sure it is never used and managed.

The factory method for test is a cheap price to pay and is at more explicit than the constructor that should never be used. More over I am sure than the real constructor never takes null as an ExecutorService.

Reply:
The main reason of this choice is that I didn't want to add code which wouldn't be needed without the proxy (the lazy loading).

Reply:
I'm not really sure I fully agree with this. It feels weird that a class in the `base/` module has a deprecated method because of CDI, something that the base module shouldn't require to function properly.